app/TagMaker.py

```diff
@@ -0,0 +1,108 @@
+from nltk.stem import PorterStemmer
+from nltk.tokenize import word_tokenize
+import nltk
+
+# nltk.download('punkt')
+
+
+class TagMaker:
+    """
+    This class takes a string text and a list of tags and can return
+    a dictionary of word stems for each tag in the tag list as well
+    as a list of words found in the text that match the tag list.
+    Any text can be searched for any version of words in the tag
+    list.
+
+    This class takes a text string and a list of possible tags.
+    With this, a dictionary of tags with their words stems for keys
+    and the original tag for values is created. The text input gets
+    converted into stems and checked  for matches in the tag
+    dictionary. A list of matching tags can be returned.
+
+    KNOWN LIMITATIONS:
+    Multi-word tags are created regardless of word order. This
+    can lead to non-correlated words in text becoming a single tag.
+
+    Abstract ideas can be described by many tags and are hard to define
+    so they can easily be missed by this simple word stem matcher.
+
+    There is a 1 for 1 relationship between tags and their stems.
+    This class does not account for multiple word stems that
+    share similar meaning.
+
+    This class was created using a pre-made nltk library. Any
+    slang, misspelled words, or colloquialism will likely be missed.
+    """
+
+    def __init__(self, text, tag_list):
+        self.text = " ".join(text.split("-"))
+        self.tag_list = tag_list
+
+    def tag_dict(self):
```

Comment:
I would recommend maybe running the stem functions in the TagList.py file. That way there is less stuff inside the TagMaker object.  Unless I'm missing something which requires for these functions to be inside the class itself.